crates/rome_js_analyze/src/utils/escape.rs

```diff
@@ -0,0 +1,80 @@
+pub fn escape<'a>(
+    unescaped_string: &'a str,
+    needs_escaping: &[&str],
+    escaping_char: char,
+) -> std::borrow::Cow<'a, str> {
+    let mut escaped = String::new();
+    let mut iter = unescaped_string.char_indices();
+    let mut is_escaped = false;
+    'unescaped: while let Some((idx, chr)) = iter.next() {
+        if chr == escaping_char {
+            is_escaped = !is_escaped;
```

Comment:
I think we need to push the `escaping_char` here to make sure that we don't drop a final `\` (which is illegal, but something our parser might produce.


```suggestion
            is_escaped = !is_escaped;
            
	        if !escaped.is_empty() {
	            escaped.push(chr);
	        }
```

Or you change it to 

```
if char == escaping_char {
	is_escaped = !is_escaped
} else {
    for candidate in needs_escaping {
	...
}

if !escaped.is_empty() {
	escaped.push(char)
}
```

Reply:
I fixed!